src/idl_gen_python.cpp

```diff
@@ -287,6 +287,62 @@ class PythonStubGenerator {
     }
   }
 
+  void GenerateObjectInitializerStub(std::stringstream &stub,
+                                     const StructDef *struct_def,
+                                     Imports *imports) const {
+    stub << "  def __init__(self";
+
+    for (const FieldDef *field : struct_def->fields.vec) {
+      if (field->deprecated) continue;
+
+      std::string field_name = namer_.Field(*field);
+      std::string field_type;
+      const Type &type = field->value.type;
+
+      if (IsScalar(type.base_type)) {
+        field_type = TypeOf(type, imports);
+        if (field->IsOptional()) { field_type += " | None"; }
```

Comment:
Note that the `|` syntax was introduced in Python 3.10, and since flatbuffers supports Python all the way back to 2.7 should be avoided.

Reply:
This is a part of `.pyi` generator. It's already used there.